### PR TITLE
TGI benchmark with llmperf

### PR DESCRIPTION
**Pull request by dacorvo**
# What does this PR do?

This adds scripts to test TGI deployments using several TGI servers on the same host and a load-balancer to achieve Data Parallelism.

The test client is [llmperf](https://github.com/ray-project/llmperf).

It also includes results for LLama 7b and Mistral v2 deployed on a inf2.48xlarge in a DP3 TP8 configuration.

### CODE REVIEW

File: benchmark/text-generation-inference/README.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+### Setup test environment
+
+```shell
+$ export LLMPerf=<path-to-llmperf>
+```
+
+### Launch benchmark run
+
+The benchmark script takes the model_id and number of concurrent users as parameters.
```
Comment (michaelbenayoun) [inline benchmark/text-generation-inference/README.md:29]:
```suggestion
The benchmark script takes the `model_id` and number of concurrent users as parameters.
```

File: benchmark/text-generation-inference/README.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+The model_id must match the one corresponding to the selected `.env` file.
```
Comment (michaelbenayoun) [inline benchmark/text-generation-inference/README.md:30]:
```suggestion
The `model_id` must match the one corresponding to the selected `.env` file.
```

File: benchmark/text-generation-inference/benchmark.sh [hunk 3]
```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
+model=${1:-NousResearch/Llama-2-7b-chat-hf}
```
Comment (michaelbenayoun) [inline benchmark/text-generation-inference/benchmark.sh:3]: Do we really want a default model for the benchmark?

File: benchmark/text-generation-inference/generate_csv.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+            "model_id": summary["model"],
+            "concurrent requests": summary["num_concurrent_requests"],
+            "throughput (t/s)": summary["results_mean_output_throughput_token_per_s"],
+            "Time-to-first-token @ P50 (s)": summary["results_ttft_s_quantiles_p50"],
+            "average latency (ms)": summary["results_inter_token_latency_s_quantiles_p50"] * 1000,
+        }
+        results.append(pd.DataFrame.from_dict(d, orient="index").transpose())
+
+df = pd.concat(results).sort_values(by="concurrent requests")
+df.to_csv("tgi-results.csv", index=False)
```
Comment (michaelbenayoun) [inline benchmark/text-generation-inference/generate_csv.py:24]: nit: I would just guard that with a `if __name__ == "__main__"`.

File: benchmark/text-generation-inference/llama-7b/.env [hunk 5]
```diff
@@ -0,0 +1,7 @@
+MODEL_ID='NousResearch/Llama-2-7b-chat-hf'
+HF_BATCH_SIZE=32
+HF_SEQUENCE_LENGTH=4096
+HF_AUTO_CAST_TYPE='fp16'
```
Comment (michaelbenayoun) [inline benchmark/text-generation-inference/llama-7b/.env:4]: Is it `fp16` or `bf16`?

**Review by michaelbenayoun — APPROVED** [review]
Just left a few questions and nits. LGTM

File: benchmark/text-generation-inference/llama-7b/.env (hunk 5 above)
Comment (dacorvo) [reply benchmark/text-generation-inference/llama-7b/.env:4]: fp16

File: benchmark/text-generation-inference/benchmark.sh (hunk 3 above)
Comment (dacorvo) [reply benchmark/text-generation-inference/benchmark.sh:3]: It also serves as an explaination of the args that can be passed

**Pull request merged**